core/src/main/groovy/noe/byteman/BytemanInstaller.groovy

```diff
@@ -0,0 +1,39 @@
+package noe.byteman
+
+import noe.common.utils.JBFile
+import noe.common.utils.PathHelper
+
+import java.nio.file.Files
+import java.nio.file.Paths
+
+/**
+ * Helping class for Byteman to handle copying byteman.jar file to visible location in the system or to user defined file.
+ */
+
+class BytemanInstaller {
+  protected static String DEFAULT_BYTEMAN_JAR_NAME = "byteman.jar"
+  protected static String DEFAULT_BYTEMAN_JAR_FOLDER = PathHelper.join(System.getProperty("java.io.tmpdir"), "noe", "byteman")
+
+  private File bytemanJar = null
+
+  BytemanInstaller(File targetBytemanJarFile = new File(DEFAULT_BYTEMAN_JAR_FOLDER, DEFAULT_BYTEMAN_JAR_NAME)) {
```

Comment:
Personally, I would be very cautious with groovy specialties in API. On Monday we talked, that support of groovy on new JDKs will require lot of changes in noe code. I don't see reason to not use standard java conventions in code we are creating. Why create new work for future?
This should be handled by telescoping constructor (we have 1 field, so it is OK).

Btw. initializing field to `null` is boilerplate. It is `null` by design.

Reply:
There's a difference though, using groovy features that require reflection, and using groovy features that are compatible with Java... Should we not use default properties, I'd suggest a telescoping constructor in this particular case.